### PR TITLE
add students behavior

### PR DESCRIPTION
**Pull request by lesiadmytrus**
Added students list
Added detail of chosen student
Added form for adding new student

### CODE REVIEW

File: src/app/app.component.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<nav class="nav">
-  <a class="nav-link active" routerLink="student">Students</a>
-  <a class="nav-link" routerLink="forms">Add</a>
+<nav class="navbar navbar-expand-lg navbar-light bg-light">
```
Comment (dmutrokot) [inline src/app/app.component.html]: Please change bg color :)

File: src/app/components/details/details.component.html [hunk 2]
```diff
@@ -1,16 +1,16 @@
 <div *ngIf="student">
 	<h2>{{ student.firstName | uppercase }} Details</h2>
 	<div>
-		<span>id: </span>{{ student._id }}
+		<span>Identity document: </span>{{ student._id }}
 	</div>
```
Comment (dmutrokot) [inline src/app/components/details/details.component.html]: Remove element with id

File: src/app/components/student/student.component.html [hunk 3]
```diff
@@ -1,11 +1,6 @@
...
   <li *ngFor="let student of students"
-    [class.selected]="student === selectedStudent"
-    (click)="onSelect(student)"
-    class="alert alert-primary" role="alert"
-    data-toggle="modal"
-    data-target="#myModal">
+    (click)="onSelect(student)">
     <span class="modal-body">{{ student.firstName }}</span>{{ student.lastName }} 
   </li>
 </ol>
```
Comment (dmutrokot) [inline src/app/components/student/student.component.html]: Please replace `ol` to `ul`

File: src/app/components/student/student.component.scss [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  cursor: pointer;
+}
+
+li:before {
+  margin-right: 10px;
+  content: counter(item);
+  background: lightblue;
+  border-radius: 100%;
+  color: white;
+  width: 1.2em;
```
Comment (dmutrokot) [inline src/app/components/student/student.component.scss]: Please use `rem`

File: src/app/components/student/student.component.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  text-align: center;
+  display: inline-block;
+}
```
Comment (dmutrokot) [inline src/app/components/student/student.component.scss]: Please use scss features

File: src/app/components/studentform/studentform.component.html [hunk 6]
```diff
@@ -1,51 +1,60 @@
...
+
+    <label><b>Email</b></label>
+    <div>
+      <input type="email" class="mail" formControlName="email">
+    </div>
+
+    <label>
+      <b>Telephone</b>
+      <input type="tel" class="tel" formControlName="phoneNumber">
+    </label><br>
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.html]: Please follow consistency between tags

File: src/app/components/studentform/studentform.component.html [hunk 7]
```diff
@@ -1,51 +1,60 @@
...
+
+    <label><b>Birthday</b>
+      <input type="date" class="birthday" formControlName="dateOfBirth">
+    </label> <br>
+
+
+    <label><b>Email</b></label>
+    <div>
+      <input type="email" class="mail" formControlName="email">
+    </div>
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.html]: Add proper usage of label

File: src/app/components/studentform/studentform.component.html [hunk 8]
```diff
@@ -1,51 +1,60 @@
...
-    <select formControlName="country">
-      <option value="Lviv">Lviv</option>
-      <option value="Kyiv">Kyiv</option>
-      <option value="Ivano-Frankivsk">Ivano-Frankivsk</option>
-      <option value="Lutsk">Lutsk</option>
-    </select>
-  </label>
-  
-  <button type="submit" (click)="add()">Submit</button>
+  <button class="registerbtn" type="submit" (click)="add()">Submit</button>
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.html]: Please use dashes for class names

File: src/app/components/studentform/studentform.component.scss [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+	background-color: white;
+}
+
+input[type=text],
+input[type=number],
+input[type=email]:focus,
+.birthday,
+.mail,
+.tel,
+.country {
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.scss]: Do not use such heavy css selectors. Add class instead

File: src/app/components/studentform/studentform.component.html (hunk 8 above)
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.html]: I guess this is not proper name for class :)

File: src/app/components/student/student.component.scss (hunk 5 above)
Comment (dmutrokot) [reply src/app/components/student/student.component.scss]: > Please use scss features

Please use nesting

File: src/app/components/studentform/studentform.component.scss [hunk 10]
```diff
@@ -1,59 +1,23 @@
...
+	input[type=email],
+	.input-style {
+  	padding: $padding;
+  	margin: 5px 0 22px 0;
+	}
+}
+
+.submit-button {
+  padding: $padding;
+  margin: 5px 0 22px 398px;
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.scss]: Please  avoid such a magic alignments as 298px

File: src/app/components/studentform/studentform.component.scss [hunk 11]
```diff
@@ -1,59 +1,23 @@
...
-	opacity: 1;
+  padding: $padding;
+	
+	.label-text {
+		font-weight: 400;
+	}
+
+	input[type=text],
+	input[type=number],
+	input[type=email],
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.scss]: Seems that you can add `inline-style` class also to these type of inputs

File: src/app/components/studentform/studentform.component.html [hunk 12]
```diff
@@ -1,60 +1,66 @@
...
+		
+    <div class="row">
+      <div class="col-md-10">        
+        <label class="label-text col-10">Country</label> 
+        <select class="custom-select">
+          <option>Italy</option>
+          <option>Denmark</option>
+          <option>Costa Rica</option>
+          <option>Colombia</option>
+          <option>Australia</option>
```
Comment (dmutrokot) [inline src/app/components/studentform/studentform.component.html]: please move country list to component/service and use ngFor

File: package.json [hunk 13]
```diff
@@ -19,7 +19,9 @@
     "@angular/platform-browser": "~7.2.0",
     "@angular/platform-browser-dynamic": "~7.2.0",
     "@angular/router": "~7.2.0",
+    "bootstrap": "^4.3.1",
     "core-js": "^2.5.4",
+    "jquery": "^3.4.1",
```
Comment (dmutrokot) [inline package.json]: Please remove `jquery` from dependencies

File: package.json [hunk 14]
```diff
@@ -44,5 +46,25 @@
...
+        "src/favicon.ico",
+        "src/assets"
+      ],
+      "styles": [
+        "src/styles.css","node_modules/bootstrap/dist/css/bootstrap.min.css"
+
+      ],
+      "scripts": ["node_modules/jquery/dist/jquery.min.js",
+                 "node_modules/bootstrap/dist/js/bootstrap.min.js"]
+    }
```
Comment (dmutrokot) [inline package.json]: Please remove `build`'s chunk of code and add `styles` prop to `angular.json` config

File: src/app/components/student-form/student-form.component.html [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+      <label class="label-text col-5">
+        First Name
+        <input type="text" class="form-control input-style" placeholder="First name" formControlName="firstName"
+          required>
+      </label>
+
+      <label class="label-text col-5">
+        Last Name
+        <input type="text" class="form-control input-style" placeholder="First name" formControlName="lastName"
+          required>
```
Comment (dmutrokot) [inline src/app/components/student-form/student-form.component.html]: I guess `required` here could be removed as you are using Angular validators

File: src/index.html [hunk 16]
```diff
@@ -7,6 +7,7 @@
 
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
```
Comment (vlodkopits) [inline src/index.html]: remove this link

File: src/app/components/student/student.component.html [hunk 17]
```diff
@@ -0,0 +1,8 @@
+<ul>
+  <li *ngFor="let student of students"
+    (click)="onSelect(student)">
```
Comment (vlodkopits) [inline src/app/components/student/student.component.html]: move `>` to new line

**Pull request merged**